confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -186,32 +308,30 @@ def get_schema(self, schema_id):
             SchemaRegistryClientError: If schema can't be found
 
         .. _Schema Registry API Reference:
-            https://docs.confluent.io/current/schema-registry/develop/api.html#get--schemas-ids-int-%20id
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--'schemas/ids/{}?format=serialized'-ids-int-%20id
 
         """  # noqa: E501
-        version = self.id_to_schema.get(schema_id, None)
+        schema = self.cache.get_schema(schema_id)
+        if schema is not None:
+            return schema
 
-        if version is not None:
-            return version
+        response = self._rest_client.GET('schemas/ids/{}'.format(schema_id))['schema']
+        schema = Schema.from_json(response)
 
-        url = url_formatter(SCHEMAS, schema_id)
-        schema = self._rest_client.GET(url, decoder=_SchemaJSONDecoder)
-        self.id_to_schema[schema_id] = schema
+        self.cache.add_schema(schema_id, schema)
 
         return schema
 
     def get_registration(self, subject_name, schema):
         """
-        POST /subjects/(string: subject)
-
         Returns ``schema`` registration information for ``subject``.
 
         Args:
-            - subject_name (str): Subject name the schema is registered under
-            - schema (Schema): Registered Schema
+            subject_name (str): Subject name the schema is registered under
+            schema (Schema): Registered Schema
```

Comment:
This is also confusing: is this a Schema instance? Or should it be a RegisteredSchema instance (as the doc string suggests)? Or a schema definition string?